### PR TITLE
Fix some UI state issues with the Perfetto iFrame

### PR DESCRIPTION
**Pull request by kenzieschmoll**

### CODE REVIEW

**Review by polina-c — APPROVED** [review]
Would test coverage be possible?

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/_perfetto_controller_web.dart [hunk 1]
```diff
@@ -249,36 +295,39 @@ class PerfettoController extends DisposableController
...
         // Once the Perfetto UI is ready, Perfetto will receive this 'PING'
         // message and return a 'PONG' message, handled in [_handleMessage].
         _postMessage(_perfettoPing);
       });
+
+      // Timeout after [_pollUntilReadyTimeout] has passed.
+      await Future.any([
+        _perfettoHandlerReady.future,
+        Future.delayed(_pollUntilReadyTimeout),
+      ]).then((_) => _pollForPerfettoHandlerReady?.cancel());
```
Comment (ditman) [inline packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/_perfetto_controller_web.dart:311]: There's [`Future.timeout` method](https://api.dart.dev/be/176281/dart-async/Future/timeout.html) to that might make this timeout pattern a little bit simpler:

```dart
try {
  await _perfettoHandlerReady.future.timeout(_pollUntilReadyTimeout);
// maybe catch the TimeoutException or define a different `onTimeout` behavior above
// so. you don't need a try-catch at all?
} finally {
  _pollForPerfettoHandlerReady?.cancel();
}
```

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/_perfetto_controller_web.dart [hunk 2]
```diff
@@ -23,13 +24,25 @@ import '../../../performance_controller.dart';
...
+var _viewIdIncrementer = 0;
+
 class PerfettoController extends DisposableController
     with AutoDisposeControllerMixin {
   PerfettoController(this.performanceController);
 
   final PerformanceController performanceController;
 
-  static const viewId = 'embedded-perfetto';
+  late final viewId = 'embedded-perfetto-${_viewIdIncrementer++}';
```
Comment (ditman) [inline packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/_perfetto_controller_web.dart:45]: sorcery-DMID1-5ojhw9r1w-369x287

**Comment by kenzieschmoll** [comment]
> Would test coverage be possible?

I will add test coverage before shipping this feature. Added TODOs to the tracking issue: https://github.com/flutter/devtools/issues/4207

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/_perfetto_controller_web.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/_perfetto_controller_web.dart]: done

**Review by CoderDake — APPROVED** [review]
LGTM :D

**Pull request merged**